Remove redundant emptiness check from genericParseNumber

Ahmad-S792 opened the pull request:
#### 408eb97e5359ee563b0bb4817598d967e75c394b
<pre>
Remove redundant emptiness check from genericParseNumber
<a href="https://bugs.webkit.org/show_bug.cgi?id=285184">https://bugs.webkit.org/show_bug.cgi?id=285184</a>
<a href="https://rdar.apple.com/142083712">rdar://142083712</a>

Reviewed by Darin Adler.

Merge: <a href="https://source.chromium.org/chromium/chromium/src/+/ab685d36561909a7ef373183f37fc3d6fc21bb83">https://source.chromium.org/chromium/chromium/src/+/ab685d36561909a7ef373183f37fc3d6fc21bb83</a>

When the &apos;start = buffer.position()&apos; condition was reached we could be sure that we
had already consumed at least one character - one of &apos;.&apos; or &apos;0&apos;-&apos;9&apos; or
whitespace (potentially also &apos;+&apos;/&apos;-&apos; as a prefix) - and that even holds
true moving the definition of |start| after consuming any leading
whitespace. The reason for this is the up-front check for any character
in the set &apos;0&apos;-&apos;9&apos; or &apos;.&apos;.
Remove the redundant checks - replacing them with an assert - while
moving the definition of &apos;start&apos; so that it doesn&apos;t point before any
leading whitespace.

* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::genericParseNumber):

Canonical link: <a href="https://commits.webkit.org/292567@main">https://commits.webkit.org/292567@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e91e2f7ed14688cd866f801ea10e6b591108607

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96451 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16065 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6096 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101521 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/46970 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98496 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16361 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24498 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73522 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30751 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99454 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12296 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87184 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53858 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12054 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/4956 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46298 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82162 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5050 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103546 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23518 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17136 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82567 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23769 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83211 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81939 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20566 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26583 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4069 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/16963 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23481 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/28636 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23140 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26620 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24881 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
Seems cleaner and harmless but unimportant. Do we have enough test coverage to be confident nothing here is incorrect?

Ahmad-S792 committed:
Remove redundant emptiness check from genericParseNumber
https://bugs.webkit.org/show_bug.cgi?id=285184
rdar://142083712

Reviewed by Darin Adler.

Merge: https://source.chromium.org/chromium/chromium/src/+/ab685d36561909a7ef373183f37fc3d6fc21bb83

When the 'start = buffer.position()' condition was reached we could be sure that we
had already consumed at least one character - one of '.' or '0'-'9' or
whitespace (potentially also '+'/'-' as a prefix) - and that even holds
true moving the definition of |start| after consuming any leading
whitespace. The reason for this is the up-front check for any character
in the set '0'-'9' or '.'.
Remove the redundant checks - replacing them with an assert - while
moving the definition of 'start' so that it doesn't point before any
leading whitespace.

* Source/WebCore/svg/SVGParserUtilities.cpp:
(WebCore::genericParseNumber):

Canonical link: https://commits.webkit.org/292567@main
